drivers/serial/uart_nrfx_uarte.c

```diff
@@ -0,0 +1,606 @@
+		nrf_uarte_hwfc_pins_set(uarte,
+					config->pselrts,
+					config->pselcts);
+	}
+
+	/* Configure flow control and parity checking */
+	nrf_uarte_configure(uarte,
+			    config->parity,
+			    config->hwfc);
+
+	err = baudrate_set(dev, config->baudrate);
+	if (err) {
+		return err;
+	}
+
+	/* Enable receiver and transmitter */
+	nrf_uarte_enable(uarte);
+
+	nrf_uarte_event_clear(uarte, NRF_UARTE_EVENT_ENDRX);
+
+	nrf_uarte_rx_buffer_set(uarte, &data->rx_data, 1);
+	nrf_uarte_task_trigger(uarte, NRF_UARTE_TASK_STARTRX);
+
+#if CONFIG_UART_INTERRUPT_DRIVEN
+	if (interrupts_active) {
+
+		irq_enable(NRFX_IRQ_NUMBER_GET(uarte));
+
+		/* Set ENDTX event by requesting fake (zero-length) transfer.
+		 * Pointer to RAM variable (data->tx_buffer) is set because
+		 * otherwise such operation may result in HardFault or RAM
+		 * corruption.
+		 */
+		nrf_uarte_tx_buffer_set(uarte, data->tx_buffer, 0);
+		nrf_uarte_task_trigger(uarte, NRF_UARTE_TASK_STARTTX);
+
+		/* switch off transmitter to save an energy */
+		nrf_uarte_task_trigger(uarte, NRF_UARTE_TASK_STOPTX);
+	}
+#endif
+
+	return 0;
+}
+
+#define UART_NRF_UARTE_DEVICE(idx)					      \
+	DEVICE_DECLARE(uart_nrfx_uarte##idx);				      \
+	static struct uarte_nrfx_data uarte_##idx##_data;		      \
+	static const struct uarte_nrfx_config uarte_##idx##_config = {	      \
+		.uarte_regs = (NRF_UARTE_Type *)NRF_UARTE##idx		      \
+	};								      \
+	static int uarte_##idx##_init(struct device *dev)		      \
+	{								      \
+		const struct uarte_init_config init_config = {		      \
+			.pseltxd = CONFIG_UART_##idx##_NRF_TX_PIN,	      \
+			.pselrxd = CONFIG_UART_##idx##_NRF_RX_PIN,	      \
+			UART_##idx##_NRF_HWFC_CONFIG			      \
+			.parity = UART_##idx##_NRF_PARITY_BIT,		      \
+			.baudrate = CONFIG_UART_##idx##_BAUD_RATE	      \
+		};							      \
+		if (UART_##idx##_INTERRUPT_DRIVEN) {			      \
+			UART_##idx##_INTERRUPTS_INIT();			      \
+		}							      \
+		return uarte_instance_init(dev,				      \
+				&init_config,				      \
+				UART_##idx##_INTERRUPT_DRIVEN);		      \
+	}								      \
+	DEVICE_AND_API_INIT(uart_nrfx_uarte##idx,			      \
+			    CONFIG_UART_##idx##_NAME,                         \
+			    uarte_##idx##_init,                               \
+			    &uarte_##idx##_data,                              \
+			    &uarte_##idx##_config,                            \
+			    PRE_KERNEL_1,                                     \
+			    CONFIG_KERNEL_INIT_PRIORITY_DEVICE,               \
+			    &uart_nrfx_uarte_driver_api)
+
+#define UART_NRF_HWFC_ENABLED(idx)                  \
+	.pselcts = CONFIG_UART_##idx##_NRF_CTS_PIN, \
+	.pselrts = CONFIG_UART_##idx##_NRF_RTS_PIN, \
+	.hwfc = NRF_UARTE_HWFC_ENABLED,
+#define UART_NRF_HWFC_DISABLED                  \
+	.pselcts = NRF_UARTE_PSEL_DISCONNECTED, \
+	.pselrts = NRF_UARTE_PSEL_DISCONNECTED, \
+	.hwfc = NRF_UARTE_HWFC_DISABLED,
+
+#define UART_NRF_IRQ_ENABLED(idx)			\
+	IRQ_CONNECT(NRFX_IRQ_NUMBER_GET(NRF_UARTE##idx),\
```

Comment:
IRQ # should come from DT.

Reply:
As above parameters, I wanted to be consistend, I am not questinging this request.

Reply:
Not sure I follow what you mean.  Consistent with what?

Reply:
With implementation in other NRF drivers like UART or SPIM.

I just created an issue for that:
https://github.com/zephyrproject-rtos/zephyr/issues/8758
Please assign me as a resolver, this will be my next task.

Reply:
Right, so any reason not to use DTS for this new driver?

Reply:
Ok, I saw @carlescufi comment about doing this for all the nRF drivers in one go, so I'm ok.

Reply:
@galak only that @jarz-nordic would prefer to do things in order: first commit the driver, then switch this and all the rest to DTS. The issue is scheduled for 1.13 and it will be Jakub's next work item after this is merged.